From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: zfcp: no port recovery after ADISC request timeout
References: bnc#487755,LTC#52565

Symptom:     The remote port remains in error state even if we receive
             a RSCN stating that the connection is re-established.
Problem:     A flag is not reset which prevents the error recovery
             to reopen the port.
Solution:    Clear the flag in question before we trigger a ERP.

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/scsi/zfcp_fc.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-sles11/drivers/s390/scsi/zfcp_fc.c
===================================================================
--- linux-sles11.orig/drivers/s390/scsi/zfcp_fc.c
+++ linux-sles11/drivers/s390/scsi/zfcp_fc.c
@@ -370,7 +370,8 @@ static void zfcp_fc_adisc_handler(unsign
 
 	if (adisc->els.status) {
 		/* request rejected or timed out */
-		zfcp_erp_port_forced_reopen(port, 0, 63, NULL);
+		zfcp_erp_port_forced_reopen(port, ZFCP_STATUS_COMMON_ERP_FAILED,
+					    63, NULL);
 		goto out;
 	}
 
